packages/lib/server/repository/team.ts

```diff
@@ -178,4 +184,185 @@ export class TeamRepository {
+        where: {
+          id: id,
+        },
+      });
+
+      const teamBilling = await TeamBilling.findAndInit(id);
+      await teamBilling.cancel();
+      return deletedTeam;
+    });
+
+    if (deletedTeam?.isOrganization && deletedTeam.slug) deleteDomain(deletedTeam.slug);
+
+    return deletedTeam;
+  }
+
+  // TODO: Move errors away from TRPC error to make it more generic
+  static async inviteMemberByToken(token: string, userId: number) {
+    const verificationToken = await prisma.verificationToken.findFirst({
+      where: {
+        token,
+        OR: [{ expiresInDays: null }, { expires: { gte: new Date() } }],
+      },
+      include: {
+        team: {
+          select: {
+            name: true,
+          },
+        },
+      },
+    });
+
+    if (!verificationToken) throw new TRPCError({ code: "NOT_FOUND", message: "Invite not found" });
+    if (!verificationToken.teamId || !verificationToken.team)
+      throw new TRPCError({
+        code: "NOT_FOUND",
+        message: "Invite token is not associated with any team",
+      });
+
+    try {
+      await prisma.membership.create({
+        data: {
+          teamId: verificationToken.teamId,
+          userId: userId,
+          role: MembershipRole.MEMBER,
+          accepted: false,
+        },
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === "P2002") {
+          throw new TRPCError({
+            code: "FORBIDDEN",
+            message: "This user is a member of this team / has a pending invitation.",
+          });
+        }
+      } else throw e;
+    }
+
+    const teamBilling = await TeamBilling.findAndInit(verificationToken.teamId);
+    await teamBilling.updateQuantity();
+
+    return verificationToken.team.name;
+  }
+
+  static async publish(teamId: number) {
+    const teamBilling = await TeamBilling.findAndInit(teamId);
+    await teamBilling.publish();
+  }
+
+  static async removeMembers(teamIds: number[], memberIds: number[], isOrg = false) {
+    const deleteMembershipPromises = [];
+
+    for (const memberId of memberIds) {
+      for (const teamId of teamIds) {
+        deleteMembershipPromises.push(
+          // This removeMember function is from @calcom/features/ee/teams/lib/removeMember.ts we should probably move it to this repository.
+          removeMember({
+            teamId,
+            memberId,
+            isOrg,
+          })
+        );
+      }
+    }
+
+    await Promise.all(deleteMembershipPromises);
+
+    const teamsBilling = await TeamBilling.findAndInitMany(teamIds);
+    const teamBillingPromises = teamsBilling.map((teamBilling) => teamBilling.updateQuantity());
+    await Promise.allSettled(teamBillingPromises);
+  }
+
+  // TODO: Add invite members by email here from inviteMember.handler.ts theres just a lot of code here that needs refactored.
+
+  private static async _deleteWorkflowRemindersOfRemovedTeam(teamId: number) {
```

Comment:
This would be better suited in a service one layer higher than the repo. 

A team repo ideally shouldn't know about how to delete workflows. It's should delete things only related to teams themselves.

If an application service one layer higher controlled this coordination of deletes, it would keep this cleaner.